src/pages/vendorpages/addProducts.tsx

```diff
@@ -0,0 +1,250 @@
+            <label className="block text-gray-700 mb-2">Quantity</label>
+            <Input
+              type="number"
+              placeholder="Enter quantity"
+              value={productQuantity.toString()}
+              onChange={(e) => setProductQuantity(parseInt(e.target.value))}
+              className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
+            />
+          </div>
+          <div className="mb-4">
+            <label className="block text-gray-700 mb-2">Discount</label>
+            <Input
+              type="number"
+              placeholder="Enter discount"
+              value={productDiscount.toString()}
+              onChange={(e) => setProductDiscount(parseInt(e.target.value))}
+              className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
+            />
+          </div>
+          <div className="mb-4">
+            <label className="block text-gray-700 mb-2">Expiry Date</label>
+            <DatePicker
+              selected={expiryDate}
+              onChange={(date) => setExpiryDate(date)}
+              dateFormat="yyyy-MM-dd"
+              className="shadow appearance-none border rounded w-[100%] py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
+              placeholderText="Select expiry date"
+            />
+          </div>
+          <div className="mb-4">
+            <label className="block text-gray-700 mb-2">Image URLs</label>
+            {[...Array(8)].map((_, i) => (
+              <Input
+                key={i}
+                type="text"
+                placeholder={`Image URL ${i + 1}`}
+                value={productImageURLs[i] || ''}
+                onChange={(e) => {
+                  const newImageURLs = [...productImageURLs];
+                  newImageURLs[i] = e.target.value;
+                  setProductImageURLs(newImageURLs);
+                }}
+                className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
+              />
+            ))}
+          </div>
+          <div className="mb-4">
+            <label className="block text-gray-700 mb-2">Description</label>
+            <textarea
+              placeholder="Enter product description"
+              value={productDescription}
+              onChange={(e) => setProductDescription(e.target.value)}
+              className="w-full p-2 border border-gray-300 rounded-md h-24 focus:outline-none focus:shadow-outline"
+            ></textarea>
+          </div>
+          <button
+            type="submit"
+            className="w-full bg-black text-white p-2 rounded-md hover:bg-gray-800"
+            disabled={isCreating}
+          >
+            {isCreating ? 'Saving changes...' : 'Save changes'}
+          </button>
```

Comment:
We have a button component that can be used or improved if it's not dynamic enough